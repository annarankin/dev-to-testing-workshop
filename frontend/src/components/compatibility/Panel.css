.Panel {
  width: 25%;
  max-width: 300px;
}

.Panel .profile-img {
  width: 100%;
  min-height: 250px;
  background-size: cover;
  background-position: center;
}
