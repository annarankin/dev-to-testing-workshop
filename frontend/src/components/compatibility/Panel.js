import React from 'react';
import './Panel.css'

const Panel = ({
  name,
  profile_image,
  caption,
  likes,
}) => {
  const profileImgStyle = {
    backgroundImage: `url('${profile_image}')`
  }

  return (
    <div className="Panel">
      <div className="profile-img" style={profileImgStyle} />

      <table>
        <thead>
          <tr><th>{name}</th></tr>
        </thead>

        <tbody>
          <tr><td>"{caption}"</td></tr>
          <tr><td>Likes: {likes.join(', ')}</td></tr>
        </tbody>
      </table>
    </div>
  )
}

export default Panel
