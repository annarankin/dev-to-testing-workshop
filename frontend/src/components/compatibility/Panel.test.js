// Provides a wrapper around a React component
// that gives us awesome testing methods!
// http://airbnb.io/enzyme/docs/api/
import { shallow } from 'enzyme';

import React from 'react';
import ReactDOM from 'react-dom';
import Panel from './Panel';

it('matches snapshot', () => {
  const props = {
    name: 'Bob the Dog',
    profile_image: 'http://image.com/dog.jpg',
    caption: 'I am a dog',
    likes: [ 'cats', 'birds', 'frisbee' ]
  }
  const componentWrapper = shallow(<Panel {...props} />)

  expect(componentWrapper).toMatchSnapshot()
});
