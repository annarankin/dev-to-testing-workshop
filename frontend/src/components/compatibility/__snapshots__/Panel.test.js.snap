// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Panel
    caption="I am a dog"
    likes={
        Array [
            "cats",
            "birds",
            "frisbee",
          ]
    }
    name="Bob the Dog"
    profile_image="http://image.com/dog.jpg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="profile-img"
          style={
                    Object {
                              "backgroundImage": "url('http://image.com/dog.jpg')",
                            }
          }
/>,
        <table>
          <thead>
                    <tr>
                              <th>
                                        Bob the Dog
                              </th>
                    </tr>
          </thead>
          <tbody>
                    <tr>
                              <td>
                                        "
                                        I am a dog
                                        "
                              </td>
                    </tr>
                    <tr>
                              <td>
                                        Likes: 
                                        cats, birds, frisbee
                              </td>
                    </tr>
          </tbody>
</table>,
      ],
      "className": "Panel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "profile-img",
          "style": Object {
            "backgroundImage": "url('http://image.com/dog.jpg')",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                            <th>
                                          Bob the Dog
                            </th>
              </tr>
</thead>,
            <tbody>
              <tr>
                            <td>
                                          "
                                          I am a dog
                                          "
                            </td>
              </tr>
              <tr>
                            <td>
                                          Likes: 
                                          cats, birds, frisbee
                            </td>
              </tr>
</tbody>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                                Bob the Dog
                </th>
</tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <th>
                  Bob the Dog
</th>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Bob the Dog",
                },
                "ref": null,
                "rendered": "Bob the Dog",
                "type": "th",
              },
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td>
                                    "
                                    I am a dog
                                    "
                  </td>
</tr>,
                <tr>
                  <td>
                                    Likes: 
                                    cats, birds, frisbee
                  </td>
</tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <td>
                    "
                    I am a dog
                    "
</td>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "\\"",
                      "I am a dog",
                      "\\"",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "\\"",
                    "I am a dog",
                    "\\"",
                  ],
                  "type": "td",
                },
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <td>
                    Likes: 
                    cats, birds, frisbee
</td>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Likes: ",
                      "cats, birds, frisbee",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Likes: ",
                    "cats, birds, frisbee",
                  ],
                  "type": "td",
                },
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="profile-img"
            style={
                        Object {
                                    "backgroundImage": "url('http://image.com/dog.jpg')",
                                  }
            }
/>,
          <table>
            <thead>
                        <tr>
                                    <th>
                                                Bob the Dog
                                    </th>
                        </tr>
            </thead>
            <tbody>
                        <tr>
                                    <td>
                                                "
                                                I am a dog
                                                "
                                    </td>
                        </tr>
                        <tr>
                                    <td>
                                                Likes: 
                                                cats, birds, frisbee
                                    </td>
                        </tr>
            </tbody>
</table>,
        ],
        "className": "Panel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "profile-img",
            "style": Object {
              "backgroundImage": "url('http://image.com/dog.jpg')",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                                <th>
                                                Bob the Dog
                                </th>
                </tr>
</thead>,
              <tbody>
                <tr>
                                <td>
                                                "
                                                I am a dog
                                                "
                                </td>
                </tr>
                <tr>
                                <td>
                                                Likes: 
                                                cats, birds, frisbee
                                </td>
                </tr>
</tbody>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                                    Bob the Dog
                  </th>
</tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <th>
                    Bob the Dog
</th>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Bob the Dog",
                  },
                  "ref": null,
                  "rendered": "Bob the Dog",
                  "type": "th",
                },
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tr>
                    <td>
                                        "
                                        I am a dog
                                        "
                    </td>
</tr>,
                  <tr>
                    <td>
                                        Likes: 
                                        cats, birds, frisbee
                    </td>
</tr>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <td>
                      "
                      I am a dog
                      "
</td>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "\\"",
                        "I am a dog",
                        "\\"",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "\\"",
                      "I am a dog",
                      "\\"",
                    ],
                    "type": "td",
                  },
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <td>
                      Likes: 
                      cats, birds, frisbee
</td>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Likes: ",
                        "cats, birds, frisbee",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Likes: ",
                      "cats, birds, frisbee",
                    ],
                    "type": "td",
                  },
                  "type": "tr",
                },
              ],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
